apps/challenges/views.py

```diff
@@ -233,6 +233,12 @@ def add_participant_team_to_challenge(
     except Challenge.DoesNotExist:
         response_data = {"error": "Challenge does not exist"}
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+    
+    if not challenge.is_registration_open:
```

Comment:
Please add tests for this condition in the API.

Reply:
@RishabhJain2018 done